lib/protobuf/json.ex

```diff
@@ -0,0 +1,99 @@
+defmodule Protobuf.JSON do
+  @moduledoc """
+  JSON encoding and decoding utilities for Protobuf structs.
+
+  It follows Google's [specs](https://developers.google.com/protocol-buffers/docs/proto3#json)
+  and reference implementation. Only `proto3` syntax is supported at the moment. Some features,
+  as [well-known](https://developers.google.com/protocol-buffers/docs/reference/google.protobuf)
+  types, are not fully supported yet.
+
+  ## Types
+
+  | proto3                   | JSON                  | Supported |
+  |--------------------------|-----------------------|-----------|
+  | `message`                | `object`              | Yes       |
+  | `enum`                   | `string`              | Yes       |
+  | `map<K,V>`               | `object`              | Yes       |
+  | `repeated V`             | `array [v, …]`        | Yes       |
+  | `bool`                   | `true, false`         | Yes       |
+  | `string`                 | `string`              | Yes       |
+  | `bytes`                  | `base64 string`       | Yes       |
+  | `int32, fixed32, uint32` | `number`              | Yes       |
+  | `int64, fixed64, uint64` | `string`              | Yes       |
+  | `float, double`          | `number`              | Yes       |
+  | `Any`                    | `object`              | No        |
+  | `Timestamp`              | `string`              | No        |
+  | `Duration`               | `string`              | No        |
+  | `Struct`                 | `object`              | No        |
+  | `Wrapper types`          | `various types`       | No        |
+  | `FieldMask`              | `string`              | No        |
+  | `ListValue`              | `array [foo, bar, …]` | No        |
+  | `Value`                  | `value`               | No        |
+  | `NullValue`              | `null`                | No        |
+  | `Empty`                  | `object`              | No        |
+
+  ## Usage
+
+  `Protobuf.JSON` requires a JSON library to work, so first make sure you have `:jason` added
+  to your dependencies:
+
+      defp deps do
+        [
+          {:jason, "~> 1.2"},
+          # ...
+        ]
+      end
+
+  With `encode/1` you can turn any `Protobuf` message struct into a JSON string:
+
+      iex> message = Car.new(color: :RED, top_speed: 125.3)
+      iex> Protobuf.JSON.encode(message)
+      {:ok, "{\\"color\\":\\"RED\\",\\"topSpeed\\":125.3}"}
+
+  JSON keys are encoded as camelCase strings, specified by the `json_name` field options. So
+  make sure to recompile the `.proto` files in your project before working with JSON encoding.
```

Comment:
We should probably mention this on the release notes.

Reply:
Should we also mention that the default is the camelcased field name? It's not too clear from the description above :)